core/domain/collection_domain_test.py

```diff
@@ -578,6 +578,27 @@ def test_delete_skill(self):
         self.assertEqual(
             collection.get_node('exp_id_1').prerequisite_skill_ids, [])
 
+    def test_get_skill_id_from_skill_name(self):
+        """Test to verify get_skill_id_from_skill_name works."""
+        collection = collection_domain.Collection.create_default_collection(
+            'exp_id')
+        self.assertEqual(collection.skills, {})
+
+        collection.add_skill('skillname1')
+        collection.add_skill('skillname2')
+        skill_id = collection.get_skill_id_from_skill_name('skillname1')
+        skills = collection.skills.keys()
+        self.assertEqual(skills[1], skill_id)
```

Comment:
I think it should be collection.skills.keys() as collection.skills is a list of collectionSkill object.

Reply:
Do you mean that collection.skills is a dict mapping skill IDs to skills? A list wouldn't have a keys() method.

If collection.skills is indeed a dict then self.assertIn(x, d) is fine, since it will check for x within the keys of the dict. (That's what "x in d" does, in Python).

Reply:
Updated.